WordPressKitTests/Utilities/URLSessionHelperTests.swift

```diff
@@ -227,6 +236,92 @@ class URLSessionHelperTests: XCTestCase {
         let expectedBody = "--\(boundary)\r\nContent-Disposition: form-data; name=\"name\"\r\n\r\nvalue\r\n--\(boundary)--\r\n"
         XCTAssertEqual(String(data: requestBody, encoding: .utf8), expectedBody)
     }
+
+    func testGetLargeData() async throws {
+        let file = try self.createLargeFile(megaBytes: 100)
+        defer {
+            try? FileManager.default.removeItem(at: file)
+        }
+
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(fileURL: file, statusCode: 200, headers: nil)
+        }
+
+        let builder = HTTPRequestBuilder(url: URL(string: "https://wordpress.org/hello")!)
+        let response = try await session.perform(request: builder, errorType: TestError.self).get()
+
+        try XCTAssertEqual(
+            sha256(XCTUnwrap(InputStream(url: file))),
+            sha256(InputStream(data: response.body))
+        )
+    }
+
+    private func createLargeFile(megaBytes: Int) throws -> URL {
+        let fileManager = FileManager.default
+        let file = try fileManager.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+            .appendingPathComponent("large-file-\(UUID().uuidString).txt")
+        fileManager.createFile(atPath: file.path, contents: nil)
+
+        let handle = try FileHandle(forUpdating: file)
+        for _ in 0..<megaBytes {
+            handle.write(Data(repeating: 46, count: 1_000_000))
+        }
+        try handle.close()
+        return file
+    }
+
+    private func sha256(_ stream: InputStream) -> SHA256Digest {
+        stream.open()
+        defer { stream.close() }
+
+        var hash = SHA256()
+        let maxLength = 1024
+        var buffer = [UInt8](repeating: 0, count: maxLength)
+        while stream.hasBytesAvailable {
+            let bytes = stream.read(&buffer, maxLength: maxLength)
+            let data = Data(bytesNoCopy: &buffer, count: bytes, deallocator: .none)
+            hash.update(data: data)
+        }
+        return hash.finalize()
+    }
+}
+
+class BackgroundURLSessionHelperTests: URLSessionHelperTests {
+
+    // swiftlint:disable weak_delegate
+    private var delegate: TestBackgroundURLSessionDelegate!
+    // swiftlint:enable weak_delegate
+
+    override func setUp() {
+        super.setUp()
+
+        delegate = TestBackgroundURLSessionDelegate()
+        session = URLSession(configuration: .default, delegate: delegate, delegateQueue: nil)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        if delegate.startedReceivingResponse {
+            XCTAssertTrue(delegate.completionCalled)
+        }
+    }
+
+}
+
+private class TestBackgroundURLSessionDelegate: BackgroundURLSessionDelegate {
+    var startedReceivingResponse = false
+    var completionCalled = false
```

Comment:
Nitpick: When building a spy I like making these `private(set)` to ensure only the spy itself can change their value and the tests cannot accidentally mess up with the state it tracks.

```suggestion
    private(set) var startedReceivingResponse = false
    private(set) var completionCalled = false
```